### PR TITLE
Add `marshal_evictions_all` table, including complete set of eviction data since 2017

### PR DESCRIPTION
**Pull request by sraby**
This PR adds a capability to have live eviction data that can update every day! Specifically, it adds a new table to nycdb that includes _all_ marshals eviction data going back to 1/1/2017 when the data is first available. We make use of an API endpoint provided by [ANHD DAP Portal](https://portal.displacementalert.org/), which is exactly what we used to gather the cleaned, geocoded, and deduplicated data for our `marshal_evictions_19` table last year. All of the code that processes the eviction data is [open source on GitHub](https://github.com/ANHD-NYC-CODE/anhd-council-backend/blob/c0b2dbb7b1c8d5fd7a8e74b320c56865bfc684c0/datasets/models/Eviction.py) (thanks to @0xStarcat!) and the specific instance of DAP Portal reloads the data from NYC Open Data every day. 

Given that we have very close ties with ANHD and have used this data in the past (@sraby has even contributed to the [address cleaning code](https://github.com/ANHD-NYC-CODE/anhd-council-backend/blob/c0b2dbb7b1c8d5fd7a8e74b320c56865bfc684c0/core/utils/address.py#L297)), this seemed like a good opportunity to utilize the pre-existing code and work to enable live, auto-updating marshal eviction data on nycdb.

Once merged, I will make sure to add to our [wiki](https://github.com/nycdb/nycdb/wiki) to explain where this data comes from.

### CODE REVIEW

File: src/nycdb/datasets/marshal_evictions.yml [hunk 1]
```diff
@@ -76,3 +79,21 @@ schema:
       MarshalLastName: text
       ResidentialCommercialInd: text
       ScheduleStatus: text
+  -
+    table_name: marshal_evictions_all
```
Comment (sraby) [inline src/nycdb/datasets/marshal_evictions.yml:83]: Does this `table_name` seem appropriate? I wanted to just name this `marshal_evictions` but that created a weird conflict where the larger dataset name and this table name were identical. What do folks think?

**Comment by sraby** [comment]
@lblok @austensen @toolness @aepyornis @0xStarcat what do you think?

**Comment by 0xStarcat** [comment]
nice!

**Review by aepyornis — APPROVED** [review]

**Comment by aepyornis** [comment]
looks great thanks!

**Pull request merged**